docarray/array/storage/base/backend.py

```diff
@@ -21,6 +23,17 @@ def _init_storage(
         **kwargs,
     ):
         self._load_offset2ids()
+        self._register_exit_hook()
+
+    def _register_exit_hook(self):
+        weakref_del = WeakMethod(self.__del__)
+
+        def exit_hook():
+            if weakref_del() is not None:
+                weakref_del()()
+
+        atexit.register(exit_hook)
+        self._deleted = False
```

Comment:
Okey, then initialize `self._deleted = False` at `__init__`

Reply:
Done. It does make it that the `__del__` and this logic get further apart in the codebase though. Previously they were in the same folder. Now they are 2 hierarchies apart